File: apps/mobile/ios/Visvine/Repositories/DirectoryRepository.swift
import Foundation

struct DirectoryRepository {
    private let api = APIClient.shared

    func getMembers(spaceId: String) async -> APIResult<[DirectoryMember]> {
        let res: APIResult<NodesResponse> = await api.request(
            "/api/data/nodes", query: ["space_id": spaceId]
        )
        switch res {
        case .success(let r):
            return .success(r.nodes.map { member in
                var copy = member
                copy.imageUrl = MediaURL.resolve(member.imageUrl)
                return copy
            })
        case .failure(let m):
            return .failure(m)
        }
    }
}

File: apps/mobile/android/app/src/main/java/com/visvine/mobile/data/repository/DirectoryRepository.kt
package com.visvine.mobile.data.repository

import com.visvine.mobile.data.model.DirectoryMember
import com.visvine.mobile.data.remote.ApiResult
import com.visvine.mobile.data.remote.MediaUrl
import com.visvine.mobile.data.remote.VisvineApi
import com.visvine.mobile.data.remote.safeApiCall
import kotlinx.serialization.json.Json
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class DirectoryRepository @Inject constructor(
    private val api: VisvineApi,
    private val json: Json,
) {
    suspend fun getMembers(spaceId: String): ApiResult<List<DirectoryMember>> =
        when (val res = safeApiCall(json) { api.getNodes(spaceId) }) {
            is ApiResult.Success -> ApiResult.Success(
                res.data.nodes.map { it.copy(imageUrl = MediaUrl.resolve(it.imageUrl)) }
            )
            is ApiResult.Failure -> res
        }
}
